Index: unix/configure
===================================================================
--- unix/configure	(revision 77749)
+++ unix/configure	(working copy)
@@ -7548,6 +7548,8 @@
 	    esac
 	    ;;
 	Darwin-*)
+	    CC_SEARCH_FLAGS='-Wl,-rpath,${LIB_RUNTIME_DIR}'
+	    LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
 	    CFLAGS_OPTIMIZE="-Os"
 	    SHLIB_CFLAGS="-fno-common"
 	    # To avoid discrepancies between what headers configure sees during
